common/models/role.js

```diff
@@ -77,14 +75,24 @@ module.exports = function(Role) {
           roles: ACL.ROLE,
         };
 
-        // resolve user model and user type if principalType is available and not "USER"
-        if (rel === 'users' &&
-          query.where.principalType && query.where.principalType !== RoleMapping.USER) {
-          relsToModels.users = this.constructor.registry.getModel(query.where.principalType);
-          relsToTypes.users = query.where.principalType;
+        var principalModel = relsToModels[rel];
+        var principalType = relsToTypes[rel];
+
+        // redefine user model and user type if user principalType is custom (available and not "USER")
+        var isCustomUserPrincipalType = rel === 'users' &&
+          query.where.principalType &&
+          query.where.principalType !== RoleMapping.USER;
+
+        if (isCustomUserPrincipalType) {
+          principalModel = this.constructor.registry.findModel(query.where.principalType);
+          principalType = query.where.principalType;
         }
 
-        listByPrincipalType(this, relsToModels[rel], relsToTypes[rel], query, callback);
+        if (principalModel) {
+          listByPrincipalType(this, principalModel, principalType, query, callback);
+        } else {
+          callback(null, null);
+        }
       };
```

Comment:
I chose here to callback(null, null) if principalModel cannot be resolved using provided principalType rather than throwing an error. this is a similar behavior as already existing in RoleMapping.user(), but the exact equivalent here would be to return empty array`[]`. Thoughts?